bedrock/src/transactions/contracts/world_gift_manager.rs

```diff
@@ -0,0 +1,314 @@
+//! This module introduces the `WorldGiftManager` contract interface.
+
+use std::{str::FromStr, sync::LazyLock};
+
+use alloy::{
+    primitives::{Address, Bytes, U256},
+    sol,
+    sol_types::SolCall,
+};
+
+use crate::{
+    primitives::PrimitiveError,
+    transactions::contracts::{
+        erc20::Erc20,
+        multisend::{MultiSend, MULTISEND_ADDRESS},
+    },
+};
+use crate::{
+    smart_account::{
+        ISafe4337Module, InstructionFlag, Is4337Encodable, NonceKeyV1, SafeOperation,
+        TransactionTypeId, UserOperation,
+    },
+    transactions::contracts::multisend::MultiSendTx,
+};
+
+pub static WORLD_GIFT_MANAGER_ADDRESS: LazyLock<Address> = LazyLock::new(|| {
+    Address::from_str("0x91479943841A4350f614Abb9745314F262F45b2e") // TODO replace with prod contract
+        .expect("failed to decode WORLD_GIFT_MANAGER_ADDRESS")
+});
+
+sol! {
+    /// The `WorldGiftManager` contract interface.
+    /// Reference: <https://github.com/worldcoin/worldcoin-gift-contracts/blob/main/src/WorldGiftManager.sol>
+    #[derive(serde::Serialize)]
+    interface IWorldGiftManager {
+        function gift(address token, uint256 giftId, address recipient, uint256 amount) external;
+        function redeem(uint256 giftId) external;
+        function cancel(uint256 giftId) external;
+    }
+}
+
+/// Enables operations with the `WorldGiftManager` contract.
+pub struct WorldGiftManagerGift {
+    /// The inner call data for the function.
+    call_data: Vec<u8>,
+    operation: u8,
+    to: Address,
+    value: U256,
+    /// gift id, randomly generated
+    gift_id: [u8; 17],
+}
+
+impl WorldGiftManagerGift {
+    pub fn new(
+        token: Address,
+        recipient: Address,
+        amount: U256,
+        gift_id: [u8; 17],
+    ) -> Self {
+        let approve_data = Erc20::encode_approve(*WORLD_GIFT_MANAGER_ADDRESS, amount);
```

Comment:
not sure how long lived or what other functionality is available in this contract, but is this unlimited approval safe?